tso-server/src/main/java/com/yahoo/omid/tso/PersistenceProcessorHandler.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright 2011-2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.omid.tso;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import org.jboss.netty.channel.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.lmax.disruptor.WorkHandler;
+import com.yahoo.omid.committable.CommitTable;
+import com.yahoo.omid.metrics.Histogram;
+import com.yahoo.omid.metrics.MetricsRegistry;
+import com.yahoo.omid.metrics.Timer;
+import com.yahoo.omid.tso.BatchPool.Batch;
+import com.yahoo.omid.tso.PersistEvent.Type;
+
+    public class PersistenceProcessorHandler
+    implements
+    WorkHandler<PersistenceProcessorImpl.PersistBatchEvent> {
+
+        private static final Logger LOG = LoggerFactory.getLogger(PersistenceProcessor.class);
+
+        private final String tsoHostAndPort;
+        private final LeaseManagement leaseManager;
+
+        final ReplyProcessor reply;
+        final RetryProcessor retryProc;
+        final CommitTable.Client commitTableClient;
+        final CommitTable.Writer writer;
+        final Panicker panicker;
+
+        final int maxBatchSize;
+
+        Batch handlerBatch;
+
+        final Timer flushTimer;
+        final Histogram batchSizeHistogram;
+
+        long lastFlush;
```

Comment:
This is not a class attribute anymore. Before it was used to calculate if the timeout had happened, but now is not used for that task anymore. It should be a local variable of the flush() method and should be called e.g. `startFlushTimeInNs`